unstructured/partition/docx.py

```diff
@@ -0,0 +1,112 @@
+
+import docx
+
+from unstructured.cleaners.core import clean_bullets
+from unstructured.documents.elements import Element, ListItem, NarrativeText, Text, Title
+from unstructured.partition.text_type import (
+    is_bulleted_text,
+    is_possible_narrative_text,
+    is_possible_title,
+)
+
+# NOTE(robinson) - documentation on built in styles can be found at the link below
+# ref: https://python-docx.readthedocs.io/en/latest/user/
+#   styles-understanding.html#paragraph-styles-in-default-template
+STYLE_TO_ELEMENT_MAPPING = {
+    "Body Text": NarrativeText,
+    "Body Text 2": NarrativeText,
+    "Body Text 3": NarrativeText,
+    "Caption": Text,  # TODO(robinson) - add caption element type
+    "Heading 1": Title,
+    "Heading 2": Title,
+    "Heading 3": Title,
+    "Heading 4": Title,
+    "Heading 5": Title,
+    "Heading 6": Title,
+    "Heading 7": Title,
+    "Heading 8": Title,
+    "Heading 9": Title,
+    "Intense Quote": Text,  # TODO(robinson) - add quote element type
+    "List": ListItem,
+    "List 2": ListItem,
+    "List 3": ListItem,
+    "List Bullet": ListItem,
+    "List Bullet 2": ListItem,
+    "List Bullet 3": ListItem,
+    "List Continue": ListItem,
+    "List Continue 2": ListItem,
+    "List Continue 3": ListItem,
+    "List Number": ListItem,
+    "List Number 2": ListItem,
+    "List Number 3": ListItem,
+    "List Paragraph": ListItem,
+    "Macro Text": Text,
+    "No Spacing": Text,
+    "Quote": Text,  # TODO(robinson) - add quote element type
+    "Subtitle": Title,
+    "TOCHeading": Title,
+    "Title": Title,
+}
+
+
+def partition_docx(filename: Optional[str] = None, file: Optional[IO] = None) -> List[Element]:
+    """Partitions Microsoft Word Documents in .docx format into its document elements.
+
+    Parameters
+    ----------
+     filename
+        A string defining the target filename path.
+    file
+        A file-like object using "rb" mode --> open(filename, "rb").
+    """
+
+    if not any([filename, file]):
+        raise ValueError("One of filename or file must be specified.")
+
+    if filename is not None and not file:
+        document = docx.Document(filename)
+    elif file is not None and not filename:
+        document = docx.Document(file)
+    else:
+        raise ValueError("Only one of filename or file can be specified.")
+
+    elements: List[Element] = []
+    for paragraph in document.paragraphs:
+        element = _paragraph_to_element(paragraph)
+        if element is not None:
+            elements.append(element)
+
+    return elements
+
+
+def _paragraph_to_element(paragraph: docx.text.paragraph.Paragraph) -> Optional[Text]:
+    """Converts a docx Paragraph object into the appropriate unstructured document element.
+    If the paragaraph style is "Normal" or unknown, we try to predict the element type from the
+    raw text."""
+    text = paragraph.text
+    style_name = paragraph.style.name
+
+    element_class = STYLE_TO_ELEMENT_MAPPING.get(style_name, None)
```

Comment:
nit: `None` is the default fallback when `get` can't find the key, so `get(style_name)` is equivalent.